frontend/pages/hosts/details/HostDetailsPage/modals/CancelActivityModal/CancelActivityModal.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useContext } from "react";
+import { noop } from "lodash";
+
+import { IHostUpcomingActivity } from "interfaces/activity";
+import activitiesAPI from "services/entities/activities";
+import { NotificationContext } from "context/notification";
+
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+
+import { upcomingActivityComponentMap } from "pages/hosts/details/cards/Activity/ActivityConfig";
+
+import { getErrorMessage } from "./helpers";
+
+const baseClass = "cancel-activity-modal";
+
+interface ICancelActivityModalProps {
+  hostId: number;
+  activity: IHostUpcomingActivity;
+  onExit: () => void;
+}
+
+const CancelActivityModal = ({
+  hostId,
+  activity,
+  onExit,
+}: ICancelActivityModalProps) => {
+  const { renderFlash } = useContext(NotificationContext);
+
+  const ActivityItemComponent = upcomingActivityComponentMap[activity.type];
+
+  const onCancelActivity = async () => {
+    try {
+      await activitiesAPI.cancelActivity(hostId, activity.uuid);
+      renderFlash("success", "Activity successfully canceled.");
+    } catch (error) {
+      // TODO: hook up error message when API is updated
+      renderFlash("error", getErrorMessage(error));
```

Comment:
Side note: Just realized we have 118 instances of "cancelled" and 74 instances of "canceled" in our repo. We should probably choose one spelling and stick to it.

Reply:
actually when i looked into this i don't see an instance of cancelled on the frontend so lets keep using cancelled going forward